Userland/Libraries/LibGfx/ImageFormats/CCITTEncoder.cpp

```diff
@@ -0,0 +1,245 @@
+{
+    return encode_from_arrays(bit_stream, length, white_make_up_codes, white_terminating_codes);
+}
+
+ErrorOr<void> encode_black_length(BigEndianOutputBitStream& bit_stream, u32 length)
+{
+    return encode_from_arrays(bit_stream, length, black_make_up_codes, black_terminating_codes);
+}
+
+ErrorOr<void> encode_horizontal_mode(BigEndianOutputBitStream& bit_stream, EncodingStatus& status, Span<Change> current_line)
+{
+    // "Detect a2."
+    auto const a2 = current_line[1];
+
+    TRY(encode_mode(bit_stream, Mode::Horizontal));
+
+    if (!status.a0.has_value() || status.a0->color == ccitt_white) {
+        auto a0_column = status.a0.map([](auto const& change) { return change.column; }).value_or(0);
+        TRY(encode_white_length(bit_stream, status.a1.column - a0_column));
+        TRY(encode_black_length(bit_stream, a2.column - status.a1.column));
+    } else {
+        TRY(encode_black_length(bit_stream, status.a1.column - status.a0->column));
+        TRY(encode_white_length(bit_stream, a2.column - status.a1.column));
+    }
+
+    // "Put a0 on a2."
+    status.a0 = a2;
+    return {};
+}
+
+}
+
+ErrorOr<void> Group4Encoder::encode(Stream& stream, Bitmap const& bitmap)
+{
+    auto bit_stream = make<BigEndianOutputBitStream>(MaybeOwned<Stream>(stream));
+
+    // 2.2.4 Coding procedure
+
+    ReferenceLine last_line;
+    // "White reference line."
+    TRY(last_line.try_empend(ccitt_black, static_cast<u32>(bitmap.width())));
+
+    for (u32 y {}; y < static_cast<u32>(bitmap.height()); ++y) {
+        auto current_line = TRY(get_reference_line(bitmap, y));
+
+        auto last_line_span = last_line.span();
+        auto current_line_span = current_line.span();
+
+        EncodingStatus status;
+
+        while (true) {
+            // "Detect a1, b1, b2."
+            update_status(status, last_line_span, current_line_span);
+
+            // "b2 to the left of a1."
+            if (status.b2.has_value() && status.b2->column < status.a1.column) {
+                // "Pass mode coding."
+                TRY(encode_pass_mode(*bit_stream, status, current_line_span));
+            } else {
+                // "|a1b1| ≤ 3"
+                if (abs(static_cast<int>(status.a1.column) - static_cast<int>(status.b1.column)) <= 3) {
+                    // "Vertical mode coding."
+                    TRY(encode_vertical_mode(*bit_stream, status));
+                } else {
+                    // "Horizontal mode coding."
+                    TRY(encode_horizontal_mode(*bit_stream, status, current_line_span));
+                }
+            }
+
+            // "End of line?"
+            if (status.a0->column >= static_cast<u32>(bitmap.width()))
+                break;
+        }
+
+        // "Reference line for next coding line."
+        last_line = move(current_line);
+    }
+
+    // EOFB.
+    TRY(bit_stream->write_bits(EOFB, 24));
+
+    // Pad bits.
+    TRY(bit_stream->align_to_byte_boundary());
+
+    return {};
+}
+
+}
```

Comment:
The trailing line break is missing

Reply:
I don't understand how it was not caught be clang-format, but anyway, done!